.github/workflows/publish.yml

```diff
@@ -0,0 +1,67 @@
+name: publish
+on:
+  workflow_dispatch:
+    inputs:
+      version-change:
+        description: Version part
+        required: true
+        type: choice
+        options:
+          - PATCH
+          - MINOR
+          - MAJOR
+
+permissions:
+  contents: write
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ github.token }}
+      VERSION_CHANGE: ${{ github.event.inputs.version-change }}
+      RELEASE_CANDIDATE: ${{ github.event.inputs.release-candidate }}
+      VERSION_FILE: src/Version.php
+      CHANGELOG_FILE: CHANGELOG.md
+    steps:
+      - uses: actions/checkout@v3
+      - run: |
+          CHANGELOG=$(cat $CHANGELOG_FILE | sed -e '/^## */,$d');
+          MAJOR=$(cat $VERSION_FILE | grep "const MAJOR" | sed -e 's/^.*\ \(=\ \)*\(\"\)*\([0-9]*\)\(\"\)*.*/\3/g');
+          MINOR=$(cat $VERSION_FILE | grep "const MINOR" | sed -e 's/^.*\ \(=\ \)*\(\"\)*\([0-9]*\)\(\"\)*.*/\3/g');
+          PATCH=$(cat $VERSION_FILE | grep "const PATCH" | sed -e 's/^.*\ \(=\ \)*\(\"\)*\([0-9]*\)\(\"\)*.*/\3/g');
+          LAST_TAG="v$MAJOR.$MINOR.$PATCH";
+          if [ "$VERSION_CHANGE" = "MAJOR" ]
```

Comment:
Major version must be manual only - it is breakable and very rare operation.
No need chance for publish new major version by misclick.